### PR TITLE
Fix #638 [MAG_DEPTHS_PLOT fails if only a single bin is produced]

### LINKED ISSUE #638: MAG_DEPTHS_PLOT fails if only a single bin is produced
### Description of the bug

If a single mag/bin is produced upstream, then making the depths heatmap fails bc it cant calculate a distance matrix.

### Command used and terminal output

```console
cmd:

nextflow run nf-core/mag \
		 -c mag.config \
		 --input mag_samplesheet.csv \
		 --outdir mag_out \
		 -work-dir mag_work \
		 -r 3.0.1 \
		 --skip_gtdbtk \
		 --skip_spades \
		 --skip_spadeshybrid \
		 --skip_concoct \
		 --kraken2_db k2_pluspf_20240112.tar.gz \
		 --genomad_db genomad_db
```

ex depths file:

```
bin     3114327 3101140 3106384 3106237 3108596
MEGAHIT-MetaBAT2-3114327.1.fa   7.610989999999999       55.530100000000004      69.84535        12.7621 21.8551
```

error:
```
Traceback (most recent call last):
  File "/home/dcallan/.nextflow/assets/nf-core/mag/bin/plot_mag_depths.py", line 83, in <module>
    sys.exit(main())
  File "/home/dcallan/.nextflow/assets/nf-core/mag/bin/plot_mag_depths.py", line 70, in main
    sns.clustermap(
  File "/usr/local/lib/python3.9/site-packages/seaborn/_decorators.py", line 46, in inner_f
    return f(**kwargs)
  File "/usr/local/lib/python3.9/site-packages/seaborn/matrix.py", line 1402, in clustermap
    return plotter.plot(metric=metric, method=method,
  File "/usr/local/lib/python3.9/site-packages/seaborn/matrix.py", line 1220, in plot
    self.plot_dendrograms(row_cluster, col_cluster, metric, method,
  File "/usr/local/lib/python3.9/site-packages/seaborn/matrix.py", line 1065, in plot_dendrograms
    self.dendrogram_row = dendrogram(
  File "/usr/local/lib/python3.9/site-packages/seaborn/_decorators.py", line 46, in inner_f
    return f(**kwargs)
  File "/usr/local/lib/python3.9/site-packages/seaborn/matrix.py", line 784, in dendrogram
    plotter = _DendrogramPlotter(data, linkage=linkage, axis=axis,
  File "/usr/local/lib/python3.9/site-packages/seaborn/matrix.py", line 594, in __init__
    self.linkage = self.calculated_linkage
  File "/usr/local/lib/python3.9/site-packages/seaborn/matrix.py", line 661, in calculated_linkage
    return self._calculate_linkage_scipy()
  File "/usr/local/lib/python3.9/site-packages/seaborn/matrix.py", line 629, in _calculate_linkage_scipy
    linkage = hierarchy.linkage(self.array, method=self.method,
  File "/usr/local/lib/python3.9/site-packages/scipy/cluster/hierarchy.py", line 1068, in linkage
    n = int(distance.num_obs_y(y))
  File "/usr/local/lib/python3.9/site-packages/scipy/spatial/distance.py", line 2572, in num_obs_y
    raise ValueError("The number of observations cannot be determined on "
ValueError: The number of observations cannot be determined on an empty distance matrix.`, size: 2148 (max: 255)
```
```


### Relevant files

_No response_

### System information

LSF Cluster using Apptainer

### PR DESCRIPTION
**Pull request by d-callan**
This PR fixes #638 by introducing a check against the number of rows in the bin depths files, and only attempting to produce heatmaps if there are > 2. 

I'll work through the checklist shortly, though may need some guidance re testing (@jfy133 )

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: subworkflows/local/depths.nf [hunk 1]
```diff
@@ -53,9 +61,10 @@ workflow DEPTHS {
         .collectFile(name:'sample_groups.tsv'){ meta, reads -> meta.id + '\t' + meta.group + '\n' }
 
     // Filter MAG depth files: use only those for plotting that contain depths for > 2 samples
+    // as well as > 2 bins
     ch_mag_depths_plot = MAG_DEPTHS.out.depths
         .map { meta, bin_depths_file ->
-            if (getColNo(bin_depths_file) > 2) [ meta, bin_depths_file ]
+            if (getColNo(bin_depths_file) > 2 && getRowNo(bin_depths_file) > 2) [ meta, bin_depths_file ]
```
Comment (prototaxites) [inline subworkflows/local/depths.nf:67]: Does the file have a header, and if so should this then be `> 3`?

Comment (d-callan) [reply subworkflows/local/depths.nf:67]: it does have a header, yes. but its actually ok to have only 2 bins... or at least, mag doesnt fail for that. this check is meant to catch the case of a single bin.

**Comment by d-callan** [comment]
so w some local testing i can confirm that : 
1. w this change we can now skip the MAG_DEPTHS_PLOT task and only that task
2. my error has resolved and mag no longer fails for my test data
3. any number of bins other than 1 does not cause failure

ive also updated changelog, etc. so i think maybe this is ready to be looked at..

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Review by jfy133 — APPROVED** [review]
LGTM, sorry it took so long for such a small PR! 

I'm going to add you to the contributors list and we are good to go :)

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: subworkflows/local/depths.nf [hunk 2]
```diff
@@ -10,6 +10,14 @@ def getColNo(filename) {
     return lines[0].split('\t').size()
 }
 
+/*
```
Comment (jfy133) [inline subworkflows/local/depths.nf:13]:
```suggestion
/*
```

**Review by jfy133 — COMMENTED** [review]
Appeasing linting...

**Pull request merged**